src/hotspot/share/services/heapDumper.cpp

```diff
@@ -748,7 +748,7 @@ class ParDumpWriter : public AbstractDumpWriter {
 
   static void before_work() {
     assert(_lock == NULL, "ParDumpWriter lock must be initialized only once");
-    _lock = new (std::nothrow) PaddedMonitor(Mutex::leaf, "Parallel HProf writer lock", Mutex::_safepoint_check_never);
+    _lock = new (std::nothrow) PaddedMonitor(Mutex::leaf, "ParallelHProfWriter_lock", Mutex::_safepoint_check_always);
```

Comment:
If you change these locks to _safepoint_check_always, you have to acquire them without the Mutex::_no_safepoint_check flags so I don't know why you don't get that assert.

Reply:
I think it may be because this is actually not a JavaThread. So the assert in `Mutex::check_no_safepoint_state` would pass. 
Moreover, I have tried to use `PaddedMonitor(Mutex::nosafepoint, "ParallelHProfWriter_lock", Mutex::_safepoint_check_never);` here, but the slowdebug would report errors as you mentioned in JDK-8274245.

Reply:
I agree the flag here and at the place for acquiring locking seems problematic. I will try to see whether I can use `Mutex::_safepoint_check_never` here and get rid of the assert.

Reply:
I agree, maybe a new issue could be created for tracking this.

Reply:
👍